sql/src/main/java/io/crate/protocols/postgres/types/IntervalType.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.postgres.types;
+
+import io.netty.buffer.ByteBuf;
+import org.joda.time.Period;
+import org.joda.time.ReadablePeriod;
+import org.joda.time.format.PeriodFormatter;
+import org.joda.time.format.PeriodFormatterBuilder;
+
+import javax.annotation.Nonnull;
+import java.nio.charset.StandardCharsets;
+
+public class IntervalType extends PGType {
+
+    private static final int OID = 1186;
+    private static final int TYPE_LEN = 16;
+    private static final int TYPE_MOD = -1;
+    public static final IntervalType INSTANCE = new IntervalType();
+    private static final PeriodFormatter PERIOD_FORMATTER = new PeriodFormatterBuilder()
+        .appendYears()
+        .appendSuffix(" year", " years")
+        .appendSeparator(" ")
+        .appendMonths()
+        .appendSuffix(" mon", " mons")
+        .appendSeparator(" ")
+        .appendDays()
+        .appendSuffix(" day", " days")
+        .appendSeparator(" ")
+        .appendHours()
+        .appendSeparator(":")
+        .appendMinutes()
+        .appendSeparator(":")
+        .appendSecondsWithOptionalMillis()
+        .toFormatter();
+
+    private IntervalType() {
+        super(OID, TYPE_LEN, TYPE_MOD, "interval");
+    }
+
+    @Override
+    public int typArray() {
+        return PGArray.INTERVAL_ARRAY.oid();
+    }
+
+    @Override
+    public int writeAsBinary(ByteBuf buffer, @Nonnull Object value) {
+        Period period = (Period) value;
+        buffer.writeInt(TYPE_LEN);
+        buffer.writeLong(period.getSeconds());
```

Comment:
See comment in the `IntervalType` about streaming. I think we need to build total values here in order to not loose information.

Reply:
Maybe extend `testByteReadWrite`  in `PGTypesTest` as part of this, so that the round-trip of encode/decode is tested.